File: Server/Server/Models/Segment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Security.Principal;
using System.Threading.Tasks;

namespace Server.Models
{
    public class Segment
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Name { get; set; }
        public int Rows { get; set; }
        public int Columns { get; set; }
        [ForeignKey("PlaneId")]
        public virtual string PlaneId { get; set; }
        public Plane Plane { get; set; }
    }
}

File: Project/src/app/models/segment.ts
export class Segment {
    id :number;
    name :string;
    rows :number
    columns:number;

    constructor(name:string, rows:number, columns:number, id:number){
        this.name=name;
        this.rows=rows;
        this.columns=columns;
        this.id=id;
    }
}
